qiskit_nature/second_q/hamiltonians/quadratic_hamiltonian.py

```diff
@@ -123,29 +123,40 @@ def __init__(
                 if not _is_antisymmetric(antisymmetric_part, rtol=rtol, atol=atol):
                     raise ValueError("Antisymmetric part must be antisymmetric.")
 
-        self._hermitian_part = hermitian_part
-        self._antisymmetric_part = antisymmetric_part
-        self._constant = constant
-
-        if self._hermitian_part is None:
-            self._hermitian_part = np.zeros((self._num_modes, self._num_modes))
-        if self._antisymmetric_part is None:
-            self._antisymmetric_part = np.zeros((self._num_modes, self._num_modes))
+        data: dict[str, float | np.ndarray] = {"": constant}
+        if hermitian_part is not None:
+            data["+-"] = hermitian_part
+        if antisymmetric_part is not None:
+            data["++"] = antisymmetric_part
+        super().__init__(data)
```

Comment:
This will always perform additional validation e.g. ensuring that the provided matrices are square (since your keys are length 2).
Since you don't do this check yourself, I think this is good. But as a user if I were to provide `validate=False`, I would also expect the validation of `PolynomialTensor` to be skipped:
```suggestion
        super().__init__(data, validate=validate)
```

Reply:
Done.